tensorflow-ops/tests/GradientTest.hs

```diff
@@ -324,6 +324,18 @@ testPad =
     V.fromList [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1] @=? dx
     V.fromList [2, 2, 3] @=? s
 
+testSlice :: Test
+testSlice =
+  testCase "testSlice" $ do
+    ([dx], [s]) <-
+      TF.runSession $ do
+        (x :: TF.Tensor TF.Value Float) <- TF.render $ TF.zeros $ TF.Shape [2, 3, 4 :: Int64]
+        (z :: TF.Tensor TF.Value Float) <- TF.render $ TF.zeros $ TF.Shape [1, 2, 2 :: Int64]
+        let y = TF.slice x (TF.constant (TF.Shape [3]) [1, 1, 1 :: Int32]) (TF.shape z)
+        calculateGradWithShape y x
+    V.fromList [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0] @=? dx
```

Comment:
This is easier to understand if it is laid out like

```haskell
let expected =
    [0, 0, 0, 0,
     0, 0, 0, 0,
     0, 0, 0, 0,

     0, 0, 0, 0,
     0, 1, 1, 0,
     0, 1, 1, 0]
V.fromList expected @=? dx
```

but we don't do that elsewhere in this file, so up to you.

Reply:
Yes, I agree with you. I've just fixed it.